src/vmm/src/config/builder.rs

```diff
@@ -123,3 +237,139 @@ impl Builder {
+        let vmm_config = Builder::default()
+            .memory_config(None as Option<&str>)
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok());
+        assert_eq!(
+            vmm_config.unwrap().memory_config,
+            MemoryConfig { size_mib: 256 }
+        );
+    }
+
+    #[test]
+    fn test_builder_vcpu_config_success() {
+        let vmm_config = Builder::default()
+            .vcpu_config(Some("num=2"))
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok());
+        assert_eq!(vmm_config.unwrap().vcpu_config, VcpuConfig { num: 2 });
+    }
+
+    #[test]
+    fn test_builder_vcpu_config_none_default() {
+        let vmm_config = Builder::default()
+            .memory_config(None as Option<&str>)
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok());
+        assert_eq!(vmm_config.unwrap().vcpu_config, VcpuConfig { num: 1 });
+    }
+
+    #[test]
+    fn test_builder_kernel_config_success_default() {
+        let vmm_config = Builder::default()
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok());
+        assert_eq!(
+            vmm_config.unwrap().kernel_config,
+            KernelConfig {
+                cmdline: DEFAULT_KERNEL_CMDLINE.to_string(),
+                himem_start: DEFAULT_HIGH_RAM_START,
+                path: PathBuf::from("bzImage")
+            }
+        );
+    }
+
+    #[test]
+    fn test_builder_kernel_config_none_error() {
+        let vmm_config = Builder::default()
+            .kernel_config(None as Option<&str>)
+            .build();
+
+        assert!(vmm_config.is_err());
+    }
+
+    #[test]
+    fn test_builder_net_config_none_default() {
+        let vmm_config = Builder::default()
+            .network_config(None as Option<&str>)
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok());
+        assert!(vmm_config.unwrap().network_config.is_none());
+    }
+
+    #[test]
+    fn test_builder_net_config_success() {
+        let vmm_config = Builder::default()
+            .network_config(Some("tap=tap0"))
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok(), "{:?}", vmm_config.err());
+        assert_eq!(
+            vmm_config.unwrap().network_config,
+            Some(NetConfig {
+                tap_name: "tap0".to_string()
+            })
+        );
+    }
+
+    #[test]
+    fn test_builder_block_config_none_default() {
+        let vmm_config = Builder::default()
+            .block_config(None as Option<&str>)
+            .kernel_config(Some("path=bzImage"))
+            .build();
+        assert!(vmm_config.is_ok());
+        assert!(vmm_config.unwrap().block_config.is_none());
+    }
+
+    #[test]
+    fn test_builder_block_config_success() {
```

Comment:
I would also add a test with all the configurations done, and check that `vmm_config` is the expected one, e.g.

```rust
fn test_builder_vmm_config_success() {
        let vmm_config = Builder::default()
            .memory_config(Some("size_mib=1024"))
            .vcpu_config(Some("num=2"))
            .net_config(Some("tap=tap0"))
            .kernel_config(Some("path=bzImage"))
            .block_config(Some("path=/dev/loop0"))
            .build();
        assert!(vmm_config.is_ok());
        assert_eq!(
            vmm_config.unwrap(),
            VMMConfig {
                memory_config: MemoryConfig { size_mib: 1024 },
                vcpu_config: VcpuConfig { num: 2 },
                kernel_config: KernelConfig {
                    cmdline: DEFAULT_KERNEL_CMDLINE.to_string(),
                    himem_start: DEFAULT_HIGH_RAM_START,
                    path: PathBuf::from("bzImage")
                },
                net_config: Some(NetConfig {
                    tap_name: "tap0".to_string()
                }),
                block_config: Some(BlockConfig {
                    path: PathBuf::from("/dev/loop0")
                })
            }
        );
}
```

Reply:
Oh!! Thanks! Totally missed this.